### PR TITLE
qlog: decimate road/wideRoad/driver encodeIdx

### PR DESCRIPTION
**Pull request by fredyshox**
Related to https://github.com/commaai/openpilot/issues/32172

### CODE REVIEW

File: cereal/services.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def __init__(self, should_log: bool, frequency: float, decimation: Optional[int]
   "pandaStates": (True, 10., 1),
   "peripheralState": (True, 2., 1),
   "radarState": (True, 20., 5),
-  "roadEncodeIdx": (False, 20., 1),
+  "roadEncodeIdx": (False, 20., 20.0),
```
Comment (sshane) [inline cereal/services.py:30]:
```suggestion
  "roadEncodeIdx": (False, 20., 20),
```

**Comment by sshane** [comment]
This will save about 80kB per qlog.

https://github.com/commaai/openpilot/pull/32723#issuecomment-2162130046

**Comment by fredyshox** [comment]
Cannot delete that, it is used in training infra

**Comment by adeebshihadeh** [comment]
> Cannot delete that, it is used in training infra

What's it used for? These are pretty massive. Can we do the aggregation in openpilot?

**Comment by fredyshox** [comment]
Its used by IsTrainable stage, which checks qlogs for frame validity, encoder lags and appropriate fps.

**Comment by adeebshihadeh** [comment]
> Its used by IsTrainable stage, which checks qlogs for frame validity, encoder lags and appropriate fps.

That check hasn't changed in years. What about doing those checks here and publishing an IsTrainable packet?

**Comment by haraschax** [comment]
I think the right solution here is to just fix this issue. We're still getting 5-10% corrupt data due to camera timing issues. If we get that down to <0.1% then we can just remove these checks from the metrics, and do these checks in the training stack.

Though on top of that it may make sense to add some kind of corruption/validity flag to the logs.

**Comment by gregjhogan** [comment]
If we do this, make sure the way we calculate the length of segments for playing video in connect (for m3u8) properly handles not having the last 19 frames (which could throw every segment length off by almost 1 second). Also, ensuring the decimation always give us the first frame time of each segment would be ideal.

**Pull request closed without merging**